rune/libenclave/agent.go

```diff
@@ -75,8 +74,19 @@ func CreateParentAgentPipe(root string, uid, gid int) (lnFile *os.File, err erro
 }
 
 func CreateChildAgentPipe(root string) (*os.File, error) {
-	path := filepath.Join(root, "agent.sock")
-	conn, err := net.Dial("unix", path)
+	// The Linux kernel only allows unix domain socket paths less than
+	// 108 bytes, so chdir() to avoid errors caused by excessively long
+	// path.
```

Comment:
Change to `Linux kernel only supports unix domain socket path with 108 characters, and thus we have to use relative path with the assist of chdir().`